### ISSUE TITLE
Consider injecting latest irb/reline on heroku console

### ISSUE THREAD

**Issue by bf4** [heroku/heroku-buildpack-ruby#1102]
since upgrading from 2.6.2 to 2.7.2 now pasting multi-line code into IRB or Pry started taking *forever*.
Ruby 2.7.2 now has [reline](https://github.com/ruby/reline) support instead of readline
and the io-console has been reworked..

when I start the console with `--legacy` I can turn off multi-line mode,
but I didn't get that working in an irbrc, probably [IRB.conf[:USE_SINGLELINE]](https://github.com/ruby/ruby/blob/v2_7_2/lib/irb/init.rb#L165-L166), 
to make the Heroku console behave.

I found a post about it https://github.com/ruby/irb/issues/43

```sh
gem install irb reline
```

works probably due to https://github.com/ruby/reline/pull/186 and related fixes.

I've since added `irb` and `reline` to my Gemfile, but I wonder if it might be worth having the buildpack inject it for console commands.

If anything else, maybe this issue will help someone :)

**Comment by schneems** [comment]
Ruby 3 just came out with major improvements to the perf (53x faster than 2.7). If you want you can specify what command `heroku run console` executes via adding a `console:` command to your Procfile:

```
console: irb --legacy
```

> If anything else, maybe this issue will help someone :)

Totally, thanks! I don't think I'm going to act on this at this time, but it will serve as good info for anyone with a similar issue.

**Comment by bf4** [comment]
FWIW, still on Ruby 2.7, but am now locking my irb/reline/io-console to a version without borked autocomplete

```ruby
gem "irb", "1.3.7"
gem "reline", "0.2.7"
gem "io-console", "0.5.9"
```


- e.g. when you try to delete in the heroku console and it takes forever or just doesn't work<img width="688" alt="Screen Shot 2022-01-11 at 4 58 01 PM" src="https://user-images.githubusercontent.com/142914/149034679-8ef3c1e2-ed72-4c0a-ba78-eb7754ed2e07.png">


Locking the gems is better than turning autocomplete altogether, e.g.

```
echo 'IRB.conf[:USE_AUTOCOMPLETE] = false' >> ~/.irbrc
```

or

```
Reline.completion_proc = nil
```

**Comment by st0012** [comment]
> Locking the gems is better than turning autocomplete altogether

Do you mean that there's something else in the new `irb` that also bothers you other than the autocomplete feature?

**Comment by bf4** [comment]
tl;dr it worked to lock the gems and now I don't worry about it
